ui-admin/src/study/surveys/ConsentView.tsx

```diff
@@ -55,32 +57,65 @@ function RawConsentView({ studyEnvContext, consent, readOnly = false }:
       readOnly={readOnly}
       onCancel={() => navigate(studyEnvFormsPath(portal.shortcode, study.shortcode, currentEnv.environmentName))}
       onSave={createNewVersion}
+      isConsentForm={true}
     />
   )
 }
 
 export type ConsentParamsT = StudyParams & {
-  consentStableId: string
+  consentStableId: string,
+  version: string
+}
+
+/** loads a consentForm */
+export const useLoadedConsentForm = (portalShortcode: string, stableId: string, version: number) => {
+  const [form, setForm] = useState<ConsentForm | undefined>()
+
+  /** load the survey from the server to get answer mappings and ensure we've got the latest content */
+  const { isLoading } = useLoadingEffect(async () => {
+    const form = await Api.getConsentForm(portalShortcode, stableId, version)
+    setForm(form)
+  }, [portalShortcode, stableId, version])
+
+  return { isLoading, form }
+}
+
+/** read consent-form-related url params */
+export const useSurveyParams = () => {
+  const params = useParams<ConsentParamsT>()
+  const [searchParams] = useSearchParams()
+  const isReadOnly = searchParams.get('readOnly') === 'true'
+  const version = params.version ? parseInt(params.version) : undefined
+
+  return { isReadOnly, version, stableId: params.consentStableId }
 }
```

Comment:
It doesn't look like this new hook ended up being used anywhere.

Regardless, I think `ConsentView` also needs to account for read-only study envs before passing `isReadOnly` into `RawConsentView `